packages/cli/src/extract_subcommands.ts

```diff
@@ -0,0 +1,8 @@
+import { Argv } from 'yargs';
+
+/**
+ * If the parser finished processing arguments, attempt extracting subcommand information.
+ */
+export const extractSubCommands = (yargs: Argv): string => {
+  return yargs.parsed ? yargs.parsed.argv._.join(' ') : '';
```

Comment:
seems like the default should be undefined so that we get the default value in the error_handler?

Reply:
The default value of `undefined` will break the shape of dimension in emitter.
[check PR thread here](https://github.com/aws-amplify/amplify-backend/pull/1356#discussion_r1579563792)

Reply:
`extractSubCommands` shouldn't assume that the only consumer is the `UsageDataEmitter`. `error_handler` should handle the case of receiving `undefined` from this method appropriately.